Mathlib/Analysis/SpecialFunctions/Trigonometric/Arctan.lean

```diff
@@ -200,6 +200,83 @@ theorem arccos_eq_arctan {x : ℝ} (h : 0 < x) : arccos x = arctan (sqrt (1 - x
   · linarith only [arcsin_pos.2 h]
 #align real.arccos_eq_arctan Real.arccos_eq_arctan
 
+theorem arctan_inv_of_pos {x : ℝ} (h : 0 < x) : arctan x⁻¹ = π / 2 - arctan x := by
+  rw [← arctan_tan (x := _ - _), tan_pi_div_two_sub, tan_arctan]
+  · norm_num; exact (arctan_lt_pi_div_two x).trans (half_lt_self_iff.mpr pi_pos)
+  · rw [sub_lt_self_iff, ← arctan_zero]; exact tanOrderIso.symm.strictMono h
+
+theorem arctan_inv_of_neg {x : ℝ} (h : x < 0) : arctan x⁻¹ = -(π / 2) - arctan x := by
+  have := arctan_inv_of_pos (neg_pos.mpr h)
+  rwa [inv_neg, arctan_neg, neg_eq_iff_eq_neg, neg_sub', arctan_neg, neg_neg] at this
+
+lemma arctan_ne_mul_pi_div_two {x : ℝ} : ∀ (k : ℤ), arctan x ≠ (2 * k + 1) * π / 2 := by
+  by_contra!
+  obtain ⟨k, h⟩ := this
+  obtain ⟨lb, ub⟩ := arctan_mem_Ioo x
+  rw [h, neg_eq_neg_one_mul, mul_div_assoc, mul_lt_mul_right (by positivity)] at lb
+  rw [h, ← one_mul (π / 2), mul_div_assoc, mul_lt_mul_right (by positivity)] at ub
+  norm_cast at lb ub; change -1 < _ at lb; omega
+
+/-- Preliminary version of `arctan_add` with more complicated hypotheses.
+Use `arctan_add` instead. -/
+theorem arctan_add' {x y : ℝ} (h₁ : -(π / 2) < arctan x + arctan y)
+    (h₂ : arctan x + arctan y < π / 2) : arctan x + arctan y = arctan ((x + y) / (1 - x * y)) := by
+  rw [← arctan_tan h₁ h₂]; congr
+  conv_rhs => rw [← tan_arctan x, ← tan_arctan y]
+  exact tan_add' ⟨arctan_ne_mul_pi_div_two, arctan_ne_mul_pi_div_two⟩
+
+lemma arctan_add_arctan_lt_pi_div_two {x y : ℝ} (h : x * y < 1) : arctan x + arctan y < π / 2 := by
+  cases' le_or_lt y 0 with hy hy
+  · rw [← add_zero (π / 2), ← arctan_zero]
+    exact add_lt_add_of_lt_of_le (arctan_lt_pi_div_two _) (tanOrderIso.symm.monotone hy)
+  rw [← lt_div_iff hy, ← inv_eq_one_div] at h
```

Comment:
Can you indent these subgoals with a focusing dot also?

Reply:
Done.